### ISSUE TITLE
Issue with the implementation of loading/saving keras models

### ISSUE THREAD

**Issue by jadevaibhav** [keras-team/keras#10454]
I have an issue with the way the saving and loading a model is implemented in Keras. When you save your model, the parameters specified while compiling are stored along with the model. I think, only model architecture and the weights should be saved as a model. So, that when loaded, the model can be compiled with new cost-function/ optimizer/ evaluation metric. This is just a suggestion. Please tell me if you have any suggestions as to this.

**Comment by brge17** [comment]
https://github.com/keras-team/keras/blob/master/keras/engine/saving.py


``` python
model.save("foo.hdf5", include_optimizer=False)
```

**Comment by jadevaibhav** [comment]
Thanks, that solved it.

**Comment by jadevaibhav** [comment]
I went through the above code, but model only including or excluding optimizer option is there. I think, this should be provided for loss and evaluation metric too. Would appreciate your comments@brge17 @fchollet

**Comment by ryanSmd** [comment]
I have the same question , and now I load the model and then compile model with new evaluation metrics or loss function.

**Comment by Dref360** [comment]
Feel free to submit a PR for the loss and metrics
